src/lib/syntax/parser.rs

```diff
@@ -757,3 +757,354 @@ impl Parser {
         self.expect(TokenData::Punctuator(p), routine)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::syntax::{
+        ast::expr::{Expr, ExprDef},
+        lexer::Lexer,
+    };
+
+    fn check_parser(js: &str, expr: &[Expr]) {
+        let mut lexer = Lexer::new(js);
+        lexer.lex().unwrap();
+
+        dbg!(Parser::new(lexer.tokens).parse_all().unwrap());
```

Comment:
Is there a reason this is wrapped in `dbg!` ?

Reply:
Sorry, it was left there from my debugging. It's removed now, and I added bitwise operators.